plugin-server/src/worker/ingestion/person-state.ts

```diff
@@ -506,10 +526,168 @@ export class PersonState {
+                SELECT
+                    helper.uuid as old_person_id,
+                    updated_ids.version,
+                    updated_ids.oldest_event
+                FROM
+                    updated_ids
+                JOIN
+                    posthog_personoverridehelper helper
+                ON
+                    helper.id = updated_ids.old_person_id;
+            `,
+            undefined,
+            'transitivePersonOverrides',
+            client
+        )
+
+        status.debug('🔁', 'person_overrides_updated', { transitiveUpdates })
+
+        const personOverrideMessages: ProducerRecord = {
+            topic: KAFKA_PERSON_OVERRIDE,
+            messages: [
+                {
+                    value: JSON.stringify({
+                        team_id: oldPerson.team_id,
+                        merged_at: castTimestampOrNow(mergedAt, TimestampFormat.ClickHouse),
+                        override_person_id: overridePerson.uuid,
+                        old_person_id: oldPerson.uuid,
+                        oldest_event: castTimestampOrNow(oldestEvent, TimestampFormat.ClickHouse),
+                        version: version,
+                    }),
+                },
+                ...transitiveUpdates.map(({ old_person_id, version, oldest_event }) => ({
+                    value: JSON.stringify({
+                        team_id: oldPerson.team_id,
+                        merged_at: castTimestampOrNow(mergedAt, TimestampFormat.ClickHouse),
+                        override_person_id: overridePerson.uuid,
+                        old_person_id: old_person_id,
+                        oldest_event: castTimestampOrNow(oldest_event, TimestampFormat.ClickHouse),
+                        version: version,
+                    }),
+                })),
+            ],
+        }
+
+        return personOverrideMessages
+    }
+
+    private async addPersonOverrideHelper(person: Person, client?: PoolClient): Promise<number> {
+        /**
+            Update the helper table that serves as a mapping between a serial ID and a Person UUID.
+
+            This mapping is used to enable an exclusion constraint in the personoverrides table, which
+            requires int[], while avoiding any constraints on "hotter" tables, like person.
+         **/
+        const {
+            rows: [{ id }],
+        } = await this.db.postgresQuery(
+            `WITH insert_id AS (
+                    INSERT INTO posthog_personoverridehelper(
+                        team_id,
+                        uuid
+                    )
+                    VALUES (
+                        ${this.teamId},
+                        '${person.uuid}'
+                    )
+                    ON CONFLICT("team_id", "uuid") DO NOTHING
+                    RETURNING id
+                )
+                SELECT * FROM insert_id
+                UNION ALL
+                -- ON CONFLICT nothing is returned, so we get the id here.
```

Comment:
we have a test for this case, right?

Reply:
The `handles a chain of overrides being applied concurrently` test has `$identify` calls that would conflict here as both of them are mapping to the same person.

Reply:
well I'm just waiting on this PR to land so I can make the changes to allow merges, don't want to create conflicts for you :)

Reply:
I am very confused by your reply. We handle this conflict in the query, it is expected to happen during normal merges.

Reply:
Of course, the merge may fail other constraints down the line, but that's not what this comment references.